### PR TITLE
Feature: csv no header

### PR DESCRIPTION
**Pull request by Matthew-Kulich**

### CODE REVIEW

File: s-pipes-modules/module-tabular/src/test/java/cz/cvut/spipes/modules/TabularModuleTest.java [hunk 1]
```diff
@@ -150,6 +148,51 @@ public void execute_TableSchemaWithNoExistingColumn_throwsException() throws URI
         assertThrows(TableSchemaException.class, () -> module.executeSelf());
     }
 
+    @Test
+    public void executeSimpleTransformationWithoutHeaders() throws URISyntaxException, IOException {
```
[the remark this answers is not in the document]
Comment (blcham) [reply s-pipes-modules/module-tabular/src/test/java/cz/cvut/spipes/modules/TabularModuleTest.java:152]: i mean something like "$Preconditions_$whatItShouldReturn"

**Review by blcham — CHANGES_REQUESTED** [review]

File: s-pipes-modules/module-tabular/src/test/java/cz/cvut/spipes/modules/TabularModuleTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Matthew-Kulich) [reply s-pipes-modules/module-tabular/src/test/java/cz/cvut/spipes/modules/TabularModuleTest.java:152]: I'm not sure, I took over the name convention from the tests from Zdenek. But of course I can name it like that

File: s-pipes-modules/module-tabular/src/main/java/cz/cvut/spipes/modules/TabularModule.java [hunk 2]
```diff
@@ -166,6 +171,32 @@ ExecutionContext executeSelf() {
...
+
+                    if(statement != null){
+                        RDFNode node = statement.getObject();
+                        RDFList rdfList = node.as(RDFList.class);
+
+                        rdfList.iterator().forEach(rdfNode -> {
+                            try {
+                                orderList.add(new URI(rdfNode.toString()));
+                            } catch (URISyntaxException e) {
+                                logError("Invalid URI: " + rdfNode);
```
[the remark this answers is not in the document]
Comment (Matthew-Kulich) [reply s-pipes-modules/module-tabular/src/main/java/cz/cvut/spipes/modules/TabularModule.java:188]: Does blank node work with JOPA?

File: s-pipes-modules/module-tabular/src/main/java/cz/cvut/spipes/modules/TabularModule.java [hunk 3]
```diff
@@ -166,6 +171,32 @@ ExecutionContext executeSelf() {
                 LOG.debug("No custom table schema found.");
             }
 
+            if(skipHeader){
```
[the remark this answers is not in the document]
Comment (Matthew-Kulich) [reply s-pipes-modules/module-tabular/src/main/java/cz/cvut/spipes/modules/TabularModule.java:174]: Agree

File: s-pipes-modules/module-tabular/src/test/resources/examples/01/input-data-schema.ttl [hunk 4]
```diff
@@ -19,34 +19,28 @@
 :bookingno_i_column   a csvw:Column;
     csvw:name "bookingno_i" ;
     csvw:property :bookingno_i ;
-    csvw:title "bookingno_i"
 .
 :Employee_No_column   a csvw:Column;
-    csvw:name "Employee_No" ;
```
Comment (blcham) [inline s-pipes-modules/module-tabular/src/test/resources/examples/01/input-data-schema.ttl:25]: @Matthew-Kulich I believe that we did not understand each other here. I believe that this file was correctly written. I believe the following is true:
- `csvw:property` does not exists in specification but I am ok to use it here. (**thoughts for future**: I believe that it would be best not even use `csvw-e:property`, but have own url e.g. "https://onto.fel.cvut.cz/csvw/"
- since csvw:property does not exists so we need to have always some kind of ID here. I guess csvw:name stands for that. Moreover csvw:title is human readable property so this is where i would expect to have " " (space) e.g. "Employee No"



-----

See: https://www.w3.org/ns/csvw

**title**
For a Transformation A natural language property that describes the format that will be generated from the transformation. For a Column: A natural language property that provides possible alternative names for the column.

-----

**name**
An atomic property that gives a single canonical name for the column. The value of this property becomes the name annotation for the described column.

File: s-pipes-modules/module-tabular/src/test/resources/examples/noHeader/schemaExample/input-data-schema.ttl [hunk 5]
```diff
@@ -19,25 +19,20 @@
 
 :Column_1   a csvw:Column;
     csvw:name "col_1";
-    csvw:title "col_1";
 .
 :Column_2   a csvw:Column;
     csvw:name "col_2" ;
-    csvw:title "col_2";
```
Comment (blcham) [inline s-pipes-modules/module-tabular/src/test/resources/examples/noHeader/schemaExample/input-data-schema.ttl]: :+1: 
I agree to remove this.

File: s-pipes-modules/module-tabular/src/test/java/cz/cvut/spipes/modules/TabularModuleTest.java [hunk 6]
```diff
@@ -107,49 +110,24 @@ public void execute_checkTableSchema() throws URISyntaxException, IOException {
 
 
 
-    @Test
-    public void execute_TableSchemaWithLessColumns_throwsException() throws URISyntaxException, IOException {
+    @DisplayName("Executes Tabular module with the different number of columns in the schema.")
+    @ParameterizedTest(name = "{index} => message=''{0} in the schema''")
+    @ValueSource(strings = {"moreColumns", "lessColumns", "noColumns"})
```
Comment (blcham) [inline s-pipes-modules/module-tabular/src/test/java/cz/cvut/spipes/modules/TabularModuleTest.java:115]: :+1 good you tried parametrized tests

**Review by blcham — CHANGES_REQUESTED** [review]
I put quite many comments, if it is too hard to follow we can have online meeting instead.

File: s-pipes-modules/module-tabular/src/main/java/cz/cvut/spipes/modules/TabularModule.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (blcham) [reply s-pipes-modules/module-tabular/src/main/java/cz/cvut/spipes/modules/TabularModule.java]: No, JOPA does not support blank nodes yet, but the specification is talking about blank nodes ... 

so we can output **non-blank** nodes, but must be able to read blank nodes.

File: s-pipes-modules/module-tabular/src/test/java/cz/cvut/spipes/modules/TabularModuleTest.java [hunk 7]
```diff
@@ -107,49 +110,24 @@ public void execute_checkTableSchema() throws URISyntaxException, IOException {
...
-        );
-
-        Model inputModel = JenaTestUtils.laodModelFromResource("/examples/04/input-data-schema.ttl");
-        module.setInputContext(ExecutionContextFactory.createContext(inputModel));
-        assertThrows(TableSchemaException.class, () -> module.executeSelf());
-    }
-
-    @Test
-    public void executeSimpleTransformationWithoutHeaders() throws URISyntaxException, IOException {
+     void executeTabularModule_withDataSchemaNoHeader_returnsNamedColumnsFromSchema()
```
[the remark this answers is not in the document]
Comment (blcham) [reply s-pipes-modules/module-tabular/src/test/java/cz/cvut/spipes/modules/TabularModuleTest.java:129]: Maybe I would even "executeSelf_withDataSchemaNoHeader" --> "executeSelfwithDataSchemaNoHeader" ... but i don't see it as very important as soon as we are consistent across all tests

[the remark this answers is not in the document]
Comment (Matthew-Kulich) [reply s-pipes-modules/module-tabular/src/test/java/cz/cvut/spipes/modules/TabularModuleTest.java:129]: Ok, It seems that we are not using underscores.

File: s-pipes-modules/module-tabular/src/test/resources/examples/01/input-data-schema.ttl (hunk 4 above)
Comment (Matthew-Kulich) [reply s-pipes-modules/module-tabular/src/test/resources/examples/01/input-data-schema.ttl:25]: Done

File: s-pipes-modules/module-tabular/src/main/java/cz/cvut/spipes/modules/TabularModule.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Matthew-Kulich) [reply s-pipes-modules/module-tabular/src/main/java/cz/cvut/spipes/modules/TabularModule.java]: This part is used only for ordering nodes but we can discuss this on standup to synchronize
